datastore.go

```diff
@@ -48,49 +48,64 @@ func (d *datastore) Put(key ds.Key, value interface{}) error {
 		return ds.ErrInvalidType
 	}
 
-	return d.DB.Set(key.Bytes(), val, 0)
+	txn := d.DB.NewTransaction(true)
+	defer txn.Discard()
+
+	err := txn.Set(key.Bytes(), val, 0)
+	if err != nil {
+		return err
+	}
+
+	//TODO: Setting callback may potentially make this faster
+	return txn.Commit(nil)
```

Comment:
Badger `Commit` docs:

```
// Commit commits the transaction, following these steps:
//
// 1. If there are no writes, return immediately.
//
// 2. Check if read rows were updated since txn started. If so, return ErrConflict.
//
// 3. If no conflict, generate a commit timestamp and update written rows' commit ts.
//
// 4. Batch up all writes, write them to value log and LSM tree.
//
// 5. If callback is provided, Badger will return immediately after checking
// for conflicts. Writes to the database will happen in the background.  If
// there is a conflict, an error will be returned and the callback will not
// run. If there are no conflicts, the callback will be called in the
// background upon successful completion of writes or any error during write.
//
// If error is nil, the transaction is successfully committed. In case of a non-nil error, the LSM
// tree won't be updated, so there's no need for any rollback.
```

What we could do is to add `unsafe` option to ds config which would pass dummy callback to commit, just logging the errors if they were to appear.

Reply:
Meh. We should be using batches when we want to be lazy.